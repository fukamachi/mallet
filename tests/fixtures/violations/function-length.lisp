;;; Test fixture for function-length rule

;; Short function - should be fine
(defun short-function (x)
  "A short function."
  (+ x 1))

;; Long function - should violate (assuming 50 line default)
(defun very-long-function (data)
  "This function is intentionally long."
  (print 1)
  (print 2)
  (print 3)
  (print 4)
  (print 5)
  (print 6)
  (print 7)
  (print 8)
  (print 9)
  (print 10)
  (print 11)
  (print 12)
  (print 13)
  (print 14)
  (print 15)
  (print 16)
  (print 17)
  (print 18)
  (print 19)
  (print 20)
  (print 21)
  (print 22)
  (print 23)
  (print 24)
  (print 25)
  (print 26)
  (print 27)
  (print 28)
  (print 29)
  (print 30)
  (print 31)
  (print 32)
  (print 33)
  (print 34)
  (print 35)
  (print 36)
  (print 37)
  (print 38)
  (print 39)
  (print 40)
  (print 41)
  (print 42)
  (print 43)
  (print 44)
  (print 45)
  (print 46)
  (print 47)
  (print 48)
  (print 49)
  (print 50)
  (print 51))

;; Another short function
(defmethod process ((x integer))
  (* x 2))
