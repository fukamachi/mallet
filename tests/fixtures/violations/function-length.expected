# Expected violations for function-length.lisp

9:0 function-length metrics
